### ISSUE TITLE
Duplicate entries in ORDER BY cause SQL error

### ISSUE THREAD

**Issue by lstanden** [NUARIG/surveyor#411]
I received the following error when attempting to render with a SQL Server backend:

```
ActiveRecord::StatementInvalid (ActiveRecord::JDBCError: A column has been specified more than once in the order by list. Columns in the order by list must be unique.: SELECT [survey_sections].* FROM [survey_sections]  WHERE [survey_sections].[survey_id] IN (1) ORDER BY display_order ASC, display_order):
```

The call which fails is in lib/surveyor/surveyor_controller_methods.rb:68

```
    @survey = Survey.with_sections.find_by_id(@response_set.survey_id)
```

As you can see from the generated SQL above, it's trying to specify the order by clause twice due to (i believe) the default scope combined with the relationship join condition statement.

**Comment by lstanden** [comment]
Removing the default_scope from SurveySection or updating the has_many and removing the condition both appear to fix this.

**Comment by rsutphin** [comment]
Thanks for the report. What version of Rails are you using?

**Comment by lstanden** [comment]
Rails 3.2.11.  We updated to the latest due to the recently SQL Injection vulnerabilities, but we haven't deployed yet.

**Comment by yoon** [comment]
Closed by 0e3e67f
